benchmark/profile_restful_api.py

```diff
@@ -28,238 +438,673 @@ def sample_requests(
+        type=str,
+        default='sharegpt',
+        choices=['sharegpt', 'random'],
+        help='Name of the dataset to benchmark on.',
+    )
+    parser.add_argument('--dataset-path',
+                        type=str,
+                        default='',
+                        help='Path to the dataset.')
+    parser.add_argument(
+        '--model',
+        type=str,
+        help='Name or path of the model. If not set, the default model will '
+        'request /v1/models for conf.',
+    )
+    parser.add_argument(
+        '--tokenizer',
+        type=str,
+        help='Name or path of the tokenizer. If not set, using the model '
+        'conf.',
+    )
+    parser.add_argument(
+        '--num-prompts',
+        type=int,
+        default=1000,
+        help='Number of prompts to process. Default is 1000.',
+    )
+    parser.add_argument(
+        '--sharegpt-output-len',
+        type=int,
+        default=None,
+        help='Output length for each request. Overrides the output length '
+        'from the ShareGPT dataset.',
+    )
+    parser.add_argument(
+        '--random-input-len',
+        type=int,
+        help='Number of input tokens per request, used only for random '
+        'dataset.',
+    )
+    parser.add_argument(
+        '--random-output-len',
+        type=int,
+        help='Number of output tokens per request, used only for random '
+        'dataset.',
+    )
+    parser.add_argument(
+        '--random-range-ratio',
+        type=float,
+        default=0.0,
+        help='Range of sampled ratio of input/output length, '
+        'used only for random dataset.',
+    )
+    parser.add_argument(
+        '--request-rate',
+        type=float,
+        default=float('inf'),
+        help='Number of requests per second. If this is inf, then all the '
+        'requests are sent at time 0. Otherwise, we use Poisson process to '
+        'synthesize the request arrival times. Default is inf.',
+    )
+    parser.add_argument('--seed', type=int, default=1, help='The random seed.')
+    parser.add_argument(
+        '--multi',
+        action='store_true',
+        help='Use request rate range rather than single value.',
+    )
+    parser.add_argument(
+        '--request-rate-range',
+        type=str,
+        default='2,34,2',
+        help='Range of request rates in the format start,stop,step. Default '
+        'is 2,34,2. It also supports a list of request rates, requiring '
+        'the parameters to not equal three.',
+    )
+    parser.add_argument('--output-file',
+                        type=str,
+                        help='Output JSONL file name.')
+    parser.add_argument(
+        '--disable-tqdm',
+        action='store_true',
+        help='Specify to disable tqdm progress bar.',
+    )
+    parser.add_argument(
+        '--disable-stream',
+        action='store_true',
+        help='Disable streaming mode.',
+    )
+    parser.add_argument(
+        '--disable-ignore-eos',
```

Comment:
Is is a kind of BC breaking for users?

Reply:
If users don't specify `--disable-ignore-eos`, it is `ignore-eos`. I think that's not BC